code/modules/shuttles/shuttle_supply_multi.dm

```diff
@@ -0,0 +1,825 @@
+
+/obj/item/elevator_contoller/dropped(mob/user)
+	. = ..()
+	UnregisterSignal(user, COMSIG_LIVING_SPEAK)
+
+/obj/item/elevator_contoller/on_enter_storage(obj/item/storage/S)
+	. = ..()
+	if(attached_to)
+		attached_to.recall_phone()
+
+/obj/item/elevator_contoller/forceMove(atom/dest)
+	. = ..()
+	if(.)
+		reset_tether()
+
+/obj/item/elevator_contoller/proc/do_zlevel_check()
+	if(!attached_to || !loc.z || !attached_to.z)
+		return FALSE
+
+	if(zlevel_transfer)
+		if(loc.z == attached_to.z)
+			zlevel_transfer = FALSE
+			if(zlevel_transfer_timer)
+				deltimer(zlevel_transfer_timer)
+			UnregisterSignal(attached_to, COMSIG_MOVABLE_MOVED)
+			return FALSE
+		return TRUE
+
+	if(attached_to && loc.z != attached_to.z)
+		zlevel_transfer = TRUE
+		zlevel_transfer_timer = addtimer(CALLBACK(src, PROC_REF(try_doing_tether)), zlevel_transfer_timeout, TIMER_UNIQUE|TIMER_STOPPABLE)
+		return TRUE
+	return FALSE
+
+
+/obj/item/elevator_contoller/proc/try_doing_tether()
+	zlevel_transfer_timer = TIMER_ID_NULL
+	zlevel_transfer = FALSE
+	UnregisterSignal(attached_to, COMSIG_MOVABLE_MOVED)
+	reset_tether()
+
+// ------------- HERE BE YE OLDE INDUSTRIAL ELEVATOR CONTROLLER MOUNT STRUCTURE ------------
+
+/obj/structure/elevator_control_mount
+	name = "Elevator Control"
+	icon = 'icons/obj/structures/structures.dmi'
+	icon_state = "elevator_control"
+	desc = "This holds the elevator controller. What a fine piece of technology."
+	pixel_y = 28
+	layer = ABOVE_MOB_LAYER
+
+	var/obj/item/elevator_contoller/attached_to
+	var/atom/tether_holder
+
+	var/range = 3
+
+	var/enabled = TRUE
+
+/obj/structure/action_relay
+	//I relay actions YO
+	var/obj/target
+
+/obj/structure/action_relay/attack_hand(mob/user)
+	if(target)
+		target.attack_hand(user)
+
+/obj/structure/elevator_control_mount/Initialize(mapload, ...)
+	. = ..()
+
+	attached_to = new /obj/item/elevator_contoller(src)
+	RegisterSignal(attached_to, COMSIG_PARENT_PREQDELETED, PROC_REF(override_delete))
+
+	for(var/obj/structure/action_relay/AR in range(10, src))
+		AR.target = src
+		AR.appearance = appearance
+		AR.pixel_y = -4
+
+/obj/structure/elevator_control_mount/update_icon()
+	. = ..()
+
+	if(attached_to in contents)
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]_hand"
+
+/obj/structure/elevator_control_mount/attack_hand(mob/user)
+	. = ..()
+
+	if(!attached_to || attached_to.loc != src)
+		return
+
+	if(!ishuman(user))
+		return
+
+	if(!enabled)
+		return
+
+	user.put_in_active_hand(attached_to)
+	update_icon()
+
+/obj/structure/elevator_control_mount/proc/recall_phone()
+	if(ismob(attached_to.loc))
+		var/mob/M = attached_to.loc
+		M.drop_held_item(attached_to)
+		playsound(get_turf(M), "rtb_handset", 100, FALSE, 7)
+
+	attached_to.forceMove(src)
+
+	update_icon()
+
+/obj/structure/elevator_control_mount/proc/override_delete()
+	SIGNAL_HANDLER
+	recall_phone()
+	return COMPONENT_ABORT_QDEL
```

Comment:
```suggestion
/obj/structure/elevator_control_mount/proc/override_delete(force)
	SIGNAL_HANDLER
	recall_phone()
	if(!force)
		return COMPONENT_ABORT_QDEL
```
this whole thing probably needs reworking (the phone code isn't that good, ik that's where you got it from) but uhh dw about it unless you really feel like it

(the phone should be made ephemeral and spawn in when you attackhand the phone, and despawn when you put it back down)